src/components/expression/expression-renderer.js

```diff
@@ -465,22 +604,51 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
       this.PROFILE_MARGINS_.LEFT = this.DEFAULT_PROFILE_MARGIN;
     }
     else {
-      this.PROFILE_MARGINS_.LEFT = this.geneLabelWidth_ + this.LABEL_MARGIN_;
+      this.PROFILE_MARGINS_.LEFT = this.geneLabelWidth_ + this.LABEL_MARGIN;
     }
     this.svgProfile_.selectAll('*').attr('display', 'inline');
   }
 
   var heatmapData = this.data.matrix;
-  this.svgProfile_.selectAll('g').remove();
+  this.svgProfile_.selectAll('*').remove();
   this.svgProfile_.attr('width', this.canvasWidth_);
 
+  var legend = this.svgProfile_.append('g')
+    .classed('legend', true)
+    .attr('transform', genotet.utils.getTransform([
+      this.PROFILE_MARGINS_.LEFT,
+      this.DEFAULT_PROFILE_LEGEND_MARGIN
+    ]));
+  var legendHeight = this.DEFAULT_PROFILE_LEGEND_HEIGHT -
+    this.DEFAULT_PROFILE_LEGEND_MARGIN;
+
+  var i = 0;
+  var profileCount = this.data.profiles.length;
+  while (i < profileCount) {
+    var geneIndex = heatmapData.geneNames
+      .indexOf(this.data.profiles[i].geneName);
```

Comment:
Usually we write chaining methods in new lines. Since this is not chaining method, I think it's better to be
```js
var geneIndex = heatmapData.geneNames.indexOf(
  this.data.profiles[i].geneName); 
```